server/service/src/sync/integrate_document.rs

```diff
@@ -95,5 +103,15 @@ fn update_encounter(con: &StorageConnection, document: &Document) -> Result<(),
         program_row,
     )
     .map_err(|err| RepositoryError::as_db_error(&format!("{:?}", err), ""))?;
+
+    update_program_events(
```

Comment:
This is only done to last_document when it's used in encounter logic:
https://vscode.dev/github/openmsupply/open-msupply/blob/1978-update-program-events-during-sync/server/service/src/programs/encounter/insert.rs#L96
and
https://github.com/openmsupply/open-msupply/blob/33be51bf979a94062275e14bf57b05eee69f2545/server/service/src/programs/encounter/update.rs#L88

What would be the difference here ? Also, was wondering for `update encounter`, what is the use case for `is_last_doc` not being true ? (since we inserting one) ?

Reply:
thanks! good spot! I remember doing some refactoring here, got interrupted and then coming back and thinking I am already done but this was obviously not the case :-(

I fixed it and added a test + mock data.

I replicated the is_last_doc logic because this helper also required the ServiceContext...